### DISCUSSION TITLE
Backport the fixes from @yut23

### DISCUSSION THREAD

**Discussion by ELLIOTTCABLE** [ssh-ident2/ssh-ident1#10] [Ideas]
This branch seems to have Python 3 support, amongst other things. Can you merge these changes upstream?

https://github.com/yut23/ssh-ident

**Comment by maddes-b** [comment]
That are quite a lot of changes in that fork. What changes are actually imporant functionalities?

ssh-ident was always working for Python 3, but defaulted to executable `python` (which is Python 2).
That was changed recently in https://github.com/ssh-ident/ssh-ident1/commit/f8bfac16a2af409fcea4da51b2ff8b922ca42f84

**Comment by maddes-b** [comment]
Using logging module was adapted with https://github.com/ssh-ident/ssh-ident1/commit/0f253ecca5b59b3cf93a6fac98e706674b037c63.

I do not see many bugs fixed:
- [Fix direction of stdout for subprocesses](https://github.com/ccontavalli/ssh-ident/commit/1dea9aa651d3bac11a0fb0e41a1d963199ca1c29)

but several enhancements:
- using shlex module methods (not available in 2.7)
- support for the NERSC sshproxy.sh
- support for TMUX
- no ssh-agent if no identity matches
